src/app/components/Carousel.tsx

```diff
@@ -0,0 +1,114 @@
+'use client';
+import { useState, useRef } from 'react';
+import Image from 'next/image';
+import { basePath } from '../utils/constants';
+import '../styles/Carousel.module.scss';
+
+export default function MultiImageCarousel() {
+  const images = [
+    `${basePath}/images/placeholder-carousel.svg`,
+    `${basePath}/images/placeholder-carousel.svg`,
+    `${basePath}/images/placeholder-carousel.svg`,
+    `${basePath}/images/placeholder-carousel.svg`,
+    `${basePath}/images/placeholder-carousel.svg`,
+    `${basePath}/images/placeholder-carousel.svg`,
+    `${basePath}/images/placeholder-carousel.svg`,
+    `${basePath}/images/placeholder-carousel.svg`,
+  ];
+
+  const [currentIndex, setCurrentIndex] = useState(0);
+  const itemsPerView = 5; // Number of images visible at once
+  const [isDragging, setIsDragging] = useState(false);
+  const [dragStartX, setDragStartX] = useState(0);
+  const [dragOffset, setDragOffset] = useState(0);
+  const [finalOffset, setFinalOffset] = useState(0); // Stores the last offset when dragging stops
+  const [offsetType, setOffsetType] = useState('%'); // Stores the last offset when dragging stops
+
+  const carouselRef = useRef(null);
+
+  // Handle drag start
+  const handleDragStart = (
+    e: React.MouseEvent<HTMLDivElement> | React.TouchEvent<HTMLDivElement>,
+  ) => {
+    setOffsetType('px');
+    const startX = 'touches' in e ? e.touches[0].clientX : e.pageX;
+    setDragStartX(startX);
+    setIsDragging(true);
+  };
+
+  // Handle dragging
+  const handleDragMove = (
+    e: React.MouseEvent<HTMLDivElement> | React.TouchEvent<HTMLDivElement>,
+  ) => {
+    if (!isDragging) return;
+
+    const currentX = 'touches' in e ? e.touches[0].clientX : e.pageX;
+    setDragOffset(currentX - dragStartX);
+  };
+
+  // Handle drag end
+  const handleDragEnd = () => {
+    setFinalOffset(finalOffset + dragOffset); // Keep track of cumulative offset
+    setIsDragging(false);
+    setDragOffset(0); // Reset drag offset after drag ends
+    // setFinalOffset(0);
+  };
+
+  const prevSlide = () => {
+    setOffsetType('%');
+    const newIndex = currentIndex === 0 ? 0 : currentIndex - 1;
+    setCurrentIndex(newIndex);
+    setFinalOffset(-newIndex * (100 / itemsPerView)); // Update finalOffset to reflect the new position
+  };
+
+  const nextSlide = () => {
+    setOffsetType('%');
+    const maxIndex = images.length - itemsPerView;
+    const newIndex = currentIndex < maxIndex ? currentIndex + 1 : maxIndex;
+    setCurrentIndex(newIndex);
+    setFinalOffset(-newIndex * (100 / itemsPerView)); // Update finalOffset to reflect the new position
+  };
+
+  return (
+    <div
```

Comment:
Do you think we should limit the width so that the carousel doesn't take up the full width of the screen? I'm on an ultrawide monitor and decided to try using my full screen and here's how it looks currently:
<img width="3196" alt="image" src="https://github.com/user-attachments/assets/70e1ea22-9f7c-440d-90f6-2d4dc58847d2">

We could wrap the component with `GridContainer` so it lines up with the rest of the page? Here's how that looks:
<img width="3200" alt="image" src="https://github.com/user-attachments/assets/181e0c52-7eae-4bd0-8761-ceecb482269f">